Sources/RegexDSL/AllRegexMatches.swift

```diff
@@ -0,0 +1,104 @@
+import Regex
+
+public struct AllRegexMatchResults<R: RegexProtocol> {
+  var input: String
+  var vm: HareVM
+  public internal(set) var startIndex: Index
+  
+  init(input: String, regex: R) {
+    let firstRegex = Regex<R.CaptureValue>(
+      ast: .concatenation([.lazyMany(.any), regex.regex.ast]))
+    let vm = HareVM(firstRegex.program.executable)
+    
+    self.startIndex = Index(vm.execute(input: input, .partialFromFront))
```

Comment:
So it might be helpful to remove this overload, making sure we're always using ranges internally.